core/hash_map.h

```diff
@@ -162,20 +162,21 @@ class HashMap {
 			new_hash_table[i] = 0;
 		}
 
-		for (int i = 0; i < (1 << hash_table_power); i++) {
+		if (hash_table) {
```

Comment:
Looks like previous code (and other methods) always assume `hash_table` to exist (the `while` doesn't event check bounds and expects them to be `hash_table_power`).
So I guess either the `if (hash_table)` can be outright removed, or more safety checks should be added (CC @karroffel).

Reply:
I don't think I ever touched this code here. That said this is a private method, so all call sites could be inspected to see if any bounds checks or checks for `NULL` are needed.